### PR TITLE
correct device pm control return value for unimplemented handlers

### PR DESCRIPTION
**Pull request by pabigot**
`device_pm_control_nop` indicates success for all operations without doing anything.  So does the weak system timer pm control handler.  Make them return an error so callers are not deceived into thinking the unmodified output parameter to `device_get_power_state()` actually represents something meaningful.

### CODE REVIEW

**Comment by pabigot** [comment]
FWIW this is so egregious a violation of expected behavior I think it's worth considering for 2.3.0 even if it is an "api change" for all the devices that don't support device power management.

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
> FWIW this is so egregious a violation of expected behavior I think it's worth considering for 2.3.0 even if it is an "api change" for all the devices that don't support device power management.

I agree. It's technically an API change but this was clearly wrong and highly unlikely that anyone relied on the old error code, so I think we can safely include this for 2.3.0.

**Comment by pabigot** [comment]
Apparently existing code including the power management infrastructure expects to see success.  Enabling device power management in conjunction with system power management with the original two patches prevented system power management from shutting anything down.  Fixed at least one failure point, but I'm withdrawing my suggestion that this should be fixed for the release.

**Review by wentongwu — APPROVED** [review]

File: tests/kernel/device/src/main.c [hunk 1]
```diff
@@ -208,6 +208,11 @@ void test_dummy_device_pm(void)
 
 	/* Set device state to DEVICE_PM_ACTIVE_STATE */
 	ret = device_set_power_state(dev, DEVICE_PM_ACTIVE_STATE, NULL, NULL);
+	if (ret == -ENOTSUP) {
+		zassert_true((ret == -ENOTSUP),
```
[the remark this answers is not in the document]
Comment (pabigot) [reply tests/kernel/device/src/main.c:212]: Yeah, I just wanted to leave some sort of assert in there, otherwise the test wouldn't check anything.

**Pull request merged**